setup.py

```diff
@@ -3,44 +3,54 @@
 from __future__ import absolute_import
 
 import io
+import os
 import re
+import sys
+import codecs
 from setuptools import setup
+from setuptools.command.test import test as TestCommand
 
 
-def read(*filenames, **kwargs):
-    encoding = kwargs.get('encoding', 'utf-8')
-    sep = kwargs.get('sep', '\n')
-    buf = []
-    for filename in filenames:
-        with io.open(filename, encoding=encoding) as f:
-            buf.append(f.read())
-    return sep.join(buf)
+rootpath = os.path.abspath(os.path.dirname(__file__))
 
 
-VERSIONFILE = "gsw/__init__.py"
-verstrline = open(VERSIONFILE, "rt").read()
-VSRE = r"^__version__ = ['\"]([^'\"]*)['\"]"
-mo = re.search(VSRE, verstrline, re.M)
-if mo:
-    verstr = mo.group(1)
-else:
-    raise RuntimeError("Unable to find version string in %s." % (VERSIONFILE,))
+class PyTest(TestCommand):
```

Comment:
It would be good to add some comments; I don't understand what is going on, what is the overall scheme.